.github/workflows/reusable-mypy.yml

```diff
@@ -0,0 +1,21 @@
+on:
+  workflow_call:
+
+jobs:
+  check-with-mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+
+      - name: Install dependencies and type information
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install mypy
```

Comment:
I'm realizing that this will always install the latest version of `mypy`, so the workflow could start failing if there are breaking changes in a new `mypy` release. We do the same thing for the [ruff action](https://github.com/ASFHyP3/actions/blob/develop/.github/workflows/reusable-ruff.yml), which surprised me since we [pin `ruff` in its-live-monitoring](https://github.com/ASFHyP3/its-live-monitoring/blob/develop/requirements-all.txt) but then as far as I can tell the pinned version is [not used by the ruff workflow in that repository](https://github.com/ASFHyP3/its-live-monitoring/blob/develop/.github/workflows/static-anaysis.yml). Also, it seems like we *do* generally use a pinned version of `cfn-lint` for our workflows, e.g:

- https://github.com/ASFHyP3/its-live-monitoring/blob/develop/.github/workflows/static-anaysis.yml
- https://github.com/ASFHyP3/hyp3/blob/develop/.github/workflows/static-analysis.yml

So, the way we're handling versions for our linting tools seems pretty inconsistent at the moment. This is probably a good opportunity to decide on our preferred way for handling that, primarily whether we want to pin those versions and manage upgrades via dependabot, or just always install the latest version.

Reply:
After some discussion we've landed on always installing the latest version of linting tools in the reusable action, and if that ends up causing problems we can always implement version pinning for those tools in each repo.